### PR TITLE
chore: add color recipe documentation

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description
Adds documentation for using color recipes.

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-components/docs/color.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+To understand the FAST color system, the first thing to understand is that all of the named colors enumerated below are algorithmic; they are a product of the *designSystem* object in which they are calculated.
+
+### Inputs
+Each color recipe expects as its sole argument the FAST *DesignSystem* object, but there are a few core pieces of data from that object that impact color resolution.
+
+### Palettes
+Each color recipe operates on a palette. A palette in an array of hexadecimal colors ordered from light to dark. By default, FAST components leverage the `neutralPalette` and the `accentPalette`. 
+
+### Background color
+This is the contextual color in which the recipe is rendering on. In foreground, outline, and divider recipes it is used the target color in which to ensure contrast against. In fill recipes it is sometimes used as the starting location in the appropriate palette to begin resolution.
```
Comment (janechu) [inline packages/web-components/fast-components/docs/color.md:18]: nit: can you update the first two sentences to be "This is the contextual color that the recipe uses to determine what color it is rendering on. In foreground, outline, and divider recipes it will use this color to ensure that it creates an accessible color that meets contrast requirements."

File: packages/web-components/fast-components/docs/color.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# Color
+FAST utilizes an adaptive color system that provides some unique advantages:
+1. Ensure text meets [WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/) contrast requirements.
+2. Easily swap from light mode to dark, or anywhere in-between.
+3. Color themeing through palette tinting.
+4. Perceptually uniform UI across different surface colors.
+
+## Algorithmic colors
```
Comment (janechu) [inline packages/web-components/fast-components/docs/color.md:8]: You use the word "recipe" multiple times below, either add the word "recipe" to the title here or the body text underneath so the reader can draw the conclusion that an algorithm that generates a color is a recipe?

File: packages/web-components/fast-components/docs/color.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+| name | Description |
+|------|-------------|
+| `neutralLayerFloating` | Used as the background for floating layers, including context menus and fly-outs. |
+| `neutralLayerCard` | Used as the background color for cards. Pair with `neutralLayerCardContainer` for the container background. |
+| `neutralLayerCardContainer` | Used as the background color for card containers. Pair with `neutralLayerCard` for the card backgrounds. |
+| `neutralLayerL1` | Used as the background color for the primary content layer (L1). |
+| `neutralLayerL1Alt` | Alternate color for L1 surfaces. |
+| `neutralLayerL2` | Used as the background for the top command surface, logically below L1. |
+| `neutralLayerL3` | Used as the background for secondary command surfaces, logically below L2. |
+| `neutralLayerL4` | Used as the background for the lowest command surface or title bar, logically below L3. |
```
Comment (janechu) [inline packages/web-components/fast-components/docs/color.md:63]: is there an accentLayer equivalent? I think we should have a short description at the top clarifying that this table belongs to the neutral palette or something so that it's understood why that extra naming is present instead of just being "Floating", "Card", "L1" etc.

File: packages/web-components/fast-components/docs/color.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+| name | Description |
+|------|-------------|
+| `neutralOutlineRest` | Used as a rest outline color for outline controls. |
+| `neutralOutlineHover` | Used as the outline of a `neutralOutline` control when hovered. |
+| `neutralOutlineActive` | Used as the outline of a `neutralOutline` control when active.  |
+| `neutralDividerRest` | Used as the color for divider elements. |
+
+#### Toggles
+Toggles have a set of recipes that are specifically tailored.
```
Comment (janechu) [inline packages/web-components/fast-components/docs/color.md:102]: Can we change this sentence a little, maybe "Toggles have a specific set of recipes."

File: packages/web-components/fast-components/docs/color.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+| `neutralFillInputSelected` | Used as the fill of the text input when selected. |
+
+#### Document Focus
+| name | Description |
+|------|-------------|
+| `neutralFocus` | The color of the focus indicator when the element has document focus. |
+| `neutralFocusInnerAccent` | The color of the inner focus-indicator when an *accent fill* element has document focus. |
+
+### Accent Recipes
+Accent recipes use the accent palette and are intended to bring attention or otherwise distinguish the element from the rest.
```
Comment (janechu) [inline packages/web-components/fast-components/docs/color.md:129]: nit: suggestion - ".. distinguish the element on the page."

File: packages/web-components/fast-components/docs/color.md (hunk 3 above)
Comment (nicholasrice) [reply packages/web-components/fast-components/docs/color.md:63]: There are not - I'll just move it under the **Neutral Recipes** heading

**Comment by nicholasrice** [comment]
@EisenbergEffect It is currently relevant for `fast-components` and `fast-components-msft`. I put this in the docs folder of `fast-components` because that package seems to be what we're surfacing in the docs site, though I'm having trouble actually *building* the docs site to see how this best incorporates.

I also expect this will be referenced or maybe even included in the *styling components* section that I also have a PR up for.

**Comment by EisenbergEffect** [comment]
@nicholasrice Before building the docs site, make sure you have built all the web-components packages. Let me know what errors you are getting. Maybe there's something from my machine that didn't make it into the PR.

**Comment by nicholasrice** [comment]
@EisenbergEffect yea I was seeing different issues on different machines - I'll grab some error logs if it happens again

File: packages/web-components/fast-components/docs/color.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
+# Color
+FAST utilizes an adaptive color system that provides some unique advantages:
+1. Ensure text meets [WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/) contrast requirements.
+2. Easily swap from light mode to dark, or anywhere in-between.
+3. Color themeing through palette tinting.
```
Comment (awentzel) [inline packages/web-components/fast-components/docs/color.md:5]: I don't understand #3.  Could you rephrase please?  Bullets may be better representative of this type of content than numbers.

File: packages/web-components/fast-components/docs/color.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+4. Perceptually uniform UI across different surface colors.
```
Comment (awentzel) [inline packages/web-components/fast-components/docs/color.md:6]: #4 is unclear.

File: packages/web-components/fast-components/docs/color.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+## Algorithmic colors
+To understand the FAST color system, the first thing to understand is that all of the named colors enumerated below are algorithmic; they are a product of the *designSystem* object in which they are calculated. In the documentation below, these algorithmic colors will be referred to as *recipes*.
```
Comment (awentzel) [inline packages/web-components/fast-components/docs/color.md:9]: This makes it more concise on the start of the paragraph.  "The first thing to understand in the FAST color system is that all ..."

File: packages/web-components/fast-components/docs/color.md (hunk 6 above)
Comment (nicholasrice) [reply packages/web-components/fast-components/docs/color.md:5]: @awentzel I'm not sure how to rephrase - to me that is pretty explanatory. Can you elaborate on what about it you don't understand? Maybe you have a suggestion?

File: packages/web-components/fast-components/docs/color.md (hunk 8 above)
Comment (nicholasrice) [reply packages/web-components/fast-components/docs/color.md]: Took most of the fluff away - try that.

**Comment by nicholasrice** [comment]
@janechu @EisenbergEffect @awentzel I changed this pretty significantly to work this into the "styling components" section of the doc site - re-requesting reviews due to the size and scope of those additions.
